### DISCUSSION TITLE
Manipulate hcl file programmatically

### DISCUSSION THREAD

**Discussion by iZnoGod722** [actions/virtual-environments#6453] [General questions]
Hi there,

I have been using this repo to build our own self hosted agents using ubuntu2004 as the base image.
As part of the build process, I've programmatically, using powershell, to add a few more custom steps to the json file before packer reads it.

I would like to now use the ubuntu2204 script, except the only problem is it's not a json file but a hcl file.

Is there a way I can some how read this file programmatically and inject my own custom steps just like how I did it with the json file.

Any help will be much appreciated.

**Comment by erik-bershel** [comment]
Hi @alan722!
[Here](https://github.com/hashicorp/hcl2) you can read about the syntax of this language, and [here](https://github.com/tmccombs/hcl2json) you can find a tool to convert to the required format.
In general, you can both read and modify the .HCL file itself for your own purposes, and convert it to .JSON, if it's more convenient for you.
